clippy_lints/src/needless_continue.rs

```diff
@@ -364,6 +368,23 @@ fn suggestion_snippet_for_continue_inside_else<'a>(cx: &EarlyContext<'_>, data:
 }
 
 fn check_and_warn<'a>(cx: &EarlyContext<'_>, expr: &'a ast::Expr) {
+    if_chain! {
+        if let ast::ExprKind::Loop(loop_block, ..) = &expr.kind;
+        if !loop_block.stmts.is_empty();
+        if let ast::StmtKind::Expr(ref statement)
+        | ast::StmtKind::Semi(ref statement) = loop_block.stmts.last().unwrap().kind;
+        if let ast::ExprKind::Continue(_) = statement.kind;
+        then {
+            span_lint_and_help(
+                cx,
+                NEEDLESS_CONTINUE,
+                loop_block.stmts.last().unwrap().span,
```

Comment:
Formatting looks off here and I don't like the unwrapping. If `loop_block.stmts.last()` already is `Some(_)` you don't have to check if it is empty first. So this could be written as:

```suggestion
        if let Some(last_stmt) = loop_block.stmts.last();
        if let StmtKind::Expr(expr) | StmtKind::Semi(expr) = &last_stmt.kind;
        if let ExprKind::Continue(_) = expr.kind;
        then {
            span_lint_and_help(
                cx,
                NEEDLESS_CONTINUE,
                last_stmt.span,
```

Also import `StmtKind` and `ExprKind` at the top of the file instead of full qualifying it here.

Reply:
I agree it is definitely better without `unwrap()`, I wasn't sure how to avoid it. 

The rust fmt just suggested to break line with pattern matching into two lines, but I also wasn't keen on the end result. Is there some other auto formatter that you use as default?

Reply:
Nope, we use `rustfmt`. But `rustfmt` (currently) can't format code in macros. Since this code is in an `if_chain!` macro, `rustfmt` doesn't know how to format it. What `rustfmt` does though is to check if a line is too long regardless if it is in a macro or not. You then get a error message something like "this line is too long, but I don't know how to format it" from `rustfmt`.